### PR TITLE
feat: add hybrid cjs-esm support

### PR DESCRIPTION
**Pull request by Nesopie**
- We discussed this in the stream but are you happy with the way tests are being done right now? Currently they're being run from the source itself. This prevents build files from getting into the bundle but I can revert to testing both of them individually as well.

- TODO: Have to change the other commands as well apart from `build` and `test`

### CODE REVIEW

**Comment by Nesopie** [comment]
I've tested out the other commands and they work fine too, @junderw can you have a look?

**Comment by junderw** [comment]
Could you convert the existing travis CI config to Github Actions CI so I can run the tests and lints on this? Thanks.

**Comment by Nesopie** [comment]
Hey! The devDependency I used to run ts files, i.e. `tsx` is not supported on lower versions of node so tests start failing on those versions (passes on lts). 

I've tried instead to run tests on two different builds without tsx but that requires you to change the way imports are done on both the files (which can't be done as they're built from the same source so you can't get two different outputs). 

One suggestion would be to just test on one build and stick with that, however I'd like your opinion on this before I move ahead

**Review by hivoltazh — APPROVED** [review]

**Review by hivoltazh — APPROVED** [review]

**Comment by junderw** [comment]
Testing one build should be fine

File: src/test/index.ts [hunk 1]
```diff
@@ -96,19 +105,19 @@ fixtures.bech32.valid.forEach((f: Fixture): void => {
   testValidFixture(f, bech32Lib.bech32);
 });
 
-fixtures.bech32.invalid.forEach((f: InvalidFixture): void => {
+((fixtures.bech32.invalid as unknown) as InvalidFixture[]).forEach((f: InvalidFixture): void => {
```
Comment (junderw) [inline src/test/index.ts:108]: I think TypeScript has a method of importing JSON files and automatically discerning the type at compile time... could you look into that?

Comment (Nesopie) [reply src/test/index.ts:108]: oh yeah, it discerned the type but the problem was that the types were wrong haha. Some fields were missing in the JSON file for the invalid fixture so I had to do this to convert the type

**Review by junderw — APPROVED** [review]

**Pull request merged**